tutorials/hoon/graph.md

```diff
@@ -0,0 +1,153 @@
++++
+title = "Graph Store"
+weight = 3
+template = "doc.html"
++++
+
+# What is Graph Store?
+
+Graph store is a gall implementation of a tree-based database that has an affordance for cyclic structures by allowing nodes to reference other nodes. It is most comparable to modern NoSQL databases in structure, though it allows for schemas to be written in the form of marks. It is used as the backend for an increasing number of Landscape applications, and is considered to be best used for social data.
```

Comment:
> Graph store is a gall implementation of a tree-based database that has an affordance for cyclic structures by allowing nodes to reference other nodes.

"Graph store" should probably either be `graph-store` or Graph Store so that its a proper noun.

I don't actually know how `graph-store` works yet, but saying it is "tree-based" and allows cyclic structures is contradictory. By definition, a tree is a connected acyclic graph. So something is off here.

Reply:
You can have nodes reference other nodes but it's a tree like most nock structures. I don't know what to call that.

Reply:
Graph store has virtual cycles (a node may reference any other node, or even multiple). But fundamentally it is storing an ordered map of nodes where each node may have children nodes, etc.

Reply:
Yeah that puts you into a bit of a bind with terminology. In that case I'd either call it a "tree with additional virtual edges", or a "finite graph with a chosen spanning tree". I think the latter is more precise but also more confusing to someone without a background in graph theory, and either way some clarification would be needed, since the term "virtual edge" is overloaded in graph theory. So let's go with virtual edges and explain it like @tacryt-socryp